src/main/kotlin/uk/gov/dluhc/printapi/config/CorrelationIdMdcConfiguration.kt

```diff
@@ -43,6 +54,46 @@ class CorrelationIdMdcInterceptor : HandlerInterceptor {
     }
 }
 
+/**
+ * Webclient exchange filter that sets the correlation ID MDC variable of either a new value, or the
+ * current value found in the MDC context. This allows for passing and logging a consistent correlation ID between
+ * disparate systems or processes using the spring Webclient.
+ */
+@Component
+class CorrelationIdMdcExchangeFilter(private val mdcExchangeFilter: MDCExchangeFilter) : ExchangeFilterFunction {
```

Comment:
This was the bit missing, that makes the Webclient correlation id aware

Reply:
Nice 👍 

Is it worth adding the equivalent bean for `RestTemplate` ? 
I know that `print-api` doesn't use `RestTemplate`, but if the idea/goal is that this class will be moved to a shared library one day (and in the immediate/short term, copy and pasted into all the other repos) .... some of our repos do use `RestTemplate` for some very specific use cases (VCA and RCA I think - possibly others too)

Reply:
Yes I didn't add the RestTemplate one here because there wasn't a RestTemplate in src/main that I could test. In OAVA we do have a RestTemplate that talks to IER I believe. I guess I could create one in src/test in the repo?

Reply:
Wont add RestTemplate interceptor as part of this PR. It'll get added in OAVA and hopefully by then we can move to a shared artefact repo.